cmake/modules/BuildQAT.cmake

```diff
@@ -0,0 +1,55 @@
+function(build_qat)
+  set(QAT_REPO https://github.com/intel/qatlib.git)
+  set(QAT_TAG "23.11.0")
+
+  include(FindMake)
+  find_make("MAKE_EXECUTABLE" "make_cmd")
+
+  set(QAT_SOURCE_DIR ${CMAKE_BINARY_DIR}/src/qatlib)
+  set(QAT_INSTALL_DIR ${QAT_SOURCE_DIR}/install)
+  set(QAT_INCLUDE_DIR ${QAT_INSTALL_DIR}/include)
+  set(QAT_LIBRARY ${QAT_INSTALL_DIR}/lib/libqat.a)
+  set(QAT_USDM_LIBRARY ${QAT_INSTALL_DIR}/lib/libusdm.a)
+
+  # this include directory won't exist until the install step, but the
+  # imported targets need it early for INTERFACE_INCLUDE_DIRECTORIES
+  file(MAKE_DIRECTORY "${QAT_INCLUDE_DIR}")
+
+  set(configure_cmd env CC=${CMAKE_C_COMPILER} ./configure --prefix=${QAT_INSTALL_DIR})
+  # disable systemd or 'make install' tries to write /usr/lib/systemd/system/qat.service
+  list(APPEND configure_cmd --disable-systemd)
+  # build a static library with -fPIC that we can link into crypto/compressor plugins
+  list(APPEND configure_cmd --with-pic --enable-static --disable-shared)
+
+  set(install_cmd ${make_cmd} install)
+  # 'make install' is missing one header that ceph requires, so copy it manually
+  list(APPEND install_cmd COMMAND cmake -E copy -t ${QAT_INCLUDE_DIR}/qat ${QAT_SOURCE_DIR}/quickassist/utilities/libusdm_drv/include/qae_mem_utils.h)
+
+  set(source_dir_args
+    SOURCE_DIR ${QAT_SOURCE_DIR}
+    GIT_REPOSITORY ${QAT_REPO}
+    GIT_TAG ${QAT_TAG}
+    GIT_SHALLOW TRUE
+    GIT_CONFIG advice.detachedHead=false)
```

Comment:
if we're enabling these by default, it probably makes more sense to add the repos as submodules and point the external projects at their source dirs